docs/core/tools/enable-tab-autocomplete.md

```diff
@@ -0,0 +1,89 @@
+
+Starting with .NET Core SDK v2.0, support for tab completion is included but disabled by default. This article describes how to enable tab completion for your terminal of choice.
+
+[!INCLUDE [topic-appliesto-net-core-2plus](~/includes/topic-appliesto-net-core-2plus.md)]
+
+Once enabled, tab completion for the .NET Core CLI is triggered by typing a `dotnet ` command in the terminal, and then hitting the TAB key. The current command line is sent to the `dotnet complete` command, and the results are processed by your terminal. You can test the results without enabling tab completion by sending something directly to the `dotnet complete` command. For example:
+
+```
+> dotnet complete "dotnet a"
+add
+clean
+--diagnostics
+migrate
+pack
+```
+
+If that command doesn't work, make sure that .NET Core 2.0 or above is installed. If it's installed, but that command still doesn't work, make sure that the `dotnet` command resolves to a version of .NET Core 2.0 and above. Use the `dotnet --version` command to see what version of `dotnet` your current path is resolving to. For more information, see [Select the .NET Core version to use](../versions/selection.md).
+
+### Examples
+
+Here are some examples of what tab completion provides:
+
+Input                                | becomes                                                                     | because
+:------------------------------------|:----------------------------------------------------------------------------|:--------------------------------
+`dotnet a⇥`                          | `dotnet add`                                                                 | `add` is the first subcommand, alphabetically.
+`dotnet add p⇥`                      | `dotnet add --help`                                                          | Tab completion matches substrings and `--help` comes first alphabetically.
+`dotnet add p⇥⇥`                    | `dotnet add package`                                                          | Pressing tab a second time brings up the next suggestion.      
+`dotnet add package Microsoft⇥`      | `dotnet add package Microsoft.ApplicationInsights.Web`                      | Results are returned alphabetically.
+`dotnet remove reference ⇥`          | `dotnet remove reference ..\..\src\OmniSharp.DotNet\OmniSharp.DotNet.csproj` | Tab completion is project file aware.
+
+## PowerShell
+
+To add tab completion to **PowerShell** for the .NET Core CLI, create or edit the profile stored in the variable `$PROFILE`. Add the following code to your profile:
+
+```powershell
+# PowerShell parameter completion shim for the dotnet CLI 
+Register-ArgumentCompleter -Native -CommandName dotnet -ScriptBlock {
+     param($commandName, $wordToComplete, $cursorPosition)
+         dotnet complete --position $cursorPosition "$wordToComplete" | ForEach-Object {
+            [System.Management.Automation.CompletionResult]::new($_, $_, 'ParameterValue', $_)
+         }
+ }
+```
+
```

Comment:
I just tested this and I'd also add the following:
- Link to https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_profiles#the-profile-variable for more information about the $profile variable
- Note that you need to allow scripts to run and link to https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_execution_policies?#change-your-execution-policy for more info